samples/net/sockets/echo_async_select/src/socket_echo_select.c

```diff
@@ -116,17 +116,18 @@ int main(void)
 		printf("error: socket(AF_INET6): %d\n", errno);
 		exit(1);
 	}
-	#ifdef IPV6_V6ONLY
+#ifndef __ZEPHYR__
 	/* For Linux, we need to make socket IPv6-only to bind it to the
 	 * same port as IPv4 socket above.
 	 */
-	int TRUE = 1;
-	res = setsockopt(serv6, IPPROTO_IPV6, IPV6_V6ONLY, &TRUE, sizeof(TRUE));
+	int optval = 1;
+	res = setsockopt(serv6, IPPROTO_IPV6, IPV6_V6ONLY, &optval,
+				sizeof(optval));
 	if (res < 0) {
-		printf("error: setsockopt: %d\n", errno);
+		perror("setsockopt");
```

Comment:
Even though this code is temporarily wrapped in "#ifndef __ZEPHYR__", it should work on Zephyr, and I'm not sure how well perror() works there (or how much code it brings and how many boards' ROM it overflows), so I suggest to leave that printf() as is.